superset/views/alerts.py

```diff
@@ -23,17 +24,61 @@
 from wtforms import BooleanField, Form, StringField
 
 from superset.constants import RouteMethod
-from superset.models.alerts import Alert, AlertLog
+from superset.models.alerts import (
+    Alert,
+    AlertLog,
+    AlertValidatorType,
+    SQLObservation,
+    SQLObserver,
+    Validator,
+)
 from superset.models.schedules import ScheduleType
 from superset.tasks.schedules import schedule_alert_query
+from superset.utils import core as utils
 from superset.utils.core import get_email_address_str, markdown
 
 from ..exceptions import SupersetException
+from ..sql_parse import ParsedQuery
+from ..tasks.alerts.oberver import check_observer_result
 from .base import SupersetModelView
 
 # TODO: access control rules for this module
 
 
+def check_observer_sql(item: "SQLObserverInlineView") -> None:
+    try:
+        parsed_query = ParsedQuery(item.sql)
+        sql = parsed_query.stripped()
+        df = item.database.get_df(sql)
```

Comment:
it may take up a lot of time, e.g. 10 min or so
let's have a test button instead that would run query & return the results or avoid running the query

Reply:
I believe having a test button on the form would require creating the form from scratch so I will remove the function and add a TODO.

Reply:
@JasonD28 in that case let's not run the query on the validation in the form